python/ray/serve/_private/router.py

```diff
@@ -568,8 +587,17 @@ async def choose_two_replicas_with_backoff(
                             get_from_all_replicas=True,
                         )
                     )
+            elif (
+                self._locality_routing
+                and backoff_index == 0
+                and len(self._colocated_replica_ids) > 0
+            ):
+                # Attempt to schedule requests to a replica on the same node in the
+                # first iteration only.
+                candidate_replica_ids = self._colocated_replica_ids
```

Comment:
Guard the approach behind the flag `locality_routing`, which is off for handles by default, but turned on for HTTP proxies.